### PR TITLE
Infrastructure Demos.

### PR DESCRIPTION
**Pull request by jyothisgm**
### Description
Infrastructure Demos.

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Review by sijeesh — COMMENTED** [review]
Looks good to me.
Can we keep "set_new_profile" and "update_from_template" tasks in one file as both use the same profile template and other resources?

File: examples/demos/create_infrastructure_with_JBOD.pp [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    ]
+  }
+}
+
+oneview_logical_enclosure{'Test LE':
+  ensure  => 'present',
+  require => Oneview_enclosure_group['Test Enclosure Group'],
+  data    => {
+    name                 => 'Test LE',
+    enclosureUris        => ['/rest/enclosures/0000000000A66101', '/rest/enclosures/0000000000A66102', '/rest/enclosures/0000000000A66103'],
```
Comment (riconem) [inline examples/demos/create_infrastructure_with_JBOD.pp:173]: Maybe we could use here a get all enclosures method or select by index.

**Comment by riconem** [comment]
Looks good to me too. :)

File: examples/demos/create_infrastructure_with_JBOD.pp (hunk 1 above)
Comment (jyothisgm) [reply examples/demos/create_infrastructure_with_JBOD.pp:173]: This is just a basic infrastructure and it is up to user to do the necessary extra steps. Moreover, the selection from an array may not accurately select the same set of enclosures every time.

**Review by sijeesh — APPROVED** [review]
LGTM

**Review by soodpr — APPROVED** [review]

**Pull request merged**